### ISSUE TITLE
(1) Incompatibility w. GregTech CE + (2) Backport waterlogged mechanic + (3) Pistons can't push fluidlogged blocks

### ISSUE THREAD

**Issue by Shazuli** [jbredwards/Fluidlogged-API#22]
***Hi***

## 1
I've noticed an incompatibility between Fluidlogged-API and [GregTech: Community Editition](https://www.curseforge.com/minecraft/mc-mods/gregtechce).
The last version the game starts is with v1.4, after that the game crash on load. In v1.4 and earlier the game starts, but when attempting to fluidlogg any block from GTCE the client crashes. Here's the crash report:
[crash-2021-07-02_11.25.43-client.txt](https://github.com/jbredwards/Fluidlogged-API/files/6753760/crash-2021-07-02_11.25.43-client.txt)

Or here as plain text:
<details>
  <summary>crash-2021-07-02_11.25.43-client.txt</summary>

```
---- Minecraft Crash Report ----

WARNING: coremods are present:
  CreativePatchingLoader (CreativeCore_v1.10.61_mc1.12.2.jar)
  SSLoadingPlugin (SereneSeasons-1.12.2-1.2.18-universal.jar)
  TransformerLoader (OpenComputers-MC1.12.2-1.7.5.192.jar)
  Quark Plugin (Quark-r1.6-179.jar)
  PerformantLoadingPlugin (performant-1.12.2-1.10.jar)
  LoadingPlugin (Bloodmoon-MC1.12.2-1.5.3.jar)
  GTCELoadingPlugin (gregtech-1.12.2-1.16.1.753.jar)
  PhosphorFMLLoadingPlugin (phosphor-1.12.2-0.2.6+build50-universal.jar)
  ParticleCullingLoadingPlugin (particleculling-1.12.2-v1.0.1.jar)
  Fluidlogged Plugin (fluidlogged_api_v1.4_mc1.12.2.jar)
  BlurPlugin (Blur-1.0.4-14.jar)
  ItemPatchingLoader (ItemPhysic_Full_1.4.34_mc1.12.2.jar)
  Thaumic Augmentation Core Plugin (ThaumicAugmentation-1.12.2-2.1.1.jar)
  ClsPlugin (CustomLoadingScreen-1.12.2-1.5.1.jar)
  FMLPlugin (Mirage-2.0.3-rc3.jar)
  EntityCullingPlugin (EntityCulling-1.12.2-2.0.5.jar)
  DynamicSurroundingsCore (DynamicSurroundings-core-1.12.2-3.5.4.3.jar)
  AdvancedRocketryPlugin (AdvancedRocketry-1.12.2-2.0.0-248-universal.jar)
  LoadingPlugin (ChunkAnimator-MC1.12-1.2.jar)
Contact their authors BEFORE contacting forge

// This is a token for 1 free hug. Redeem at your nearest Mojangsta: [~~HUG~~]

Time: 7/2/21 11:25 AM
Description: Unexpected error

java.lang.NullPointerException: Unexpected error
	at git.jbredwards.fluidlogged_api.util.FluidloggedUtils.tryFluidlogBlock(FluidloggedUtils.java:154)
	at git.jbredwards.fluidlogged_api.common.event.FluidloggedEvents.runBucketFull(FluidloggedEvents.java:209)
	at git.jbredwards.fluidlogged_api.common.event.FluidloggedEvents.fluidPlaceOrTake(FluidloggedEvents.java:174)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_105_FluidloggedEvents_fluidPlaceOrTake_FillBucketEvent.invoke(.dynamic)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182)
	at net.minecraftforge.event.ForgeEventFactory.onBucketUse(ForgeEventFactory.java:469)
	at net.minecraftforge.fluids.UniversalBucket.func_77659_a(UniversalBucket.java:164)
	at net.minecraft.item.ItemStack.func_77957_a(ItemStack.java:217)
	at net.minecraft.client.multiplayer.PlayerControllerMP.func_187101_a(PlayerControllerMP.java:511)
	at net.minecraft.client.Minecraft.func_147121_ag(Minecraft.java:1597)
	at net.minecraft.client.Minecraft.func_184117_aA(Minecraft.java:2254)
	at net.minecraft.client.Minecraft.func_184118_az(Minecraft.java:2020)
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1808)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1098)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:398)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:196)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:231)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at git.jbredwards.fluidlogged_api.util.FluidloggedUtils.tryFluidlogBlock(FluidloggedUtils.java:154)
	at git.jbredwards.fluidlogged_api.common.event.FluidloggedEvents.runBucketFull(FluidloggedEvents.java:209)
	at git.jbredwards.fluidlogged_api.common.event.FluidloggedEvents.fluidPlaceOrTake(FluidloggedEvents.java:174)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_105_FluidloggedEvents_fluidPlaceOrTake_FillBucketEvent.invoke(.dynamic)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182)
	at net.minecraftforge.event.ForgeEventFactory.onBucketUse(ForgeEventFactory.java:469)
	at net.minecraftforge.fluids.UniversalBucket.func_77659_a(UniversalBucket.java:164)
	at net.minecraft.item.ItemStack.func_77957_a(ItemStack.java:217)
	at net.minecraft.client.multiplayer.PlayerControllerMP.func_187101_a(PlayerControllerMP.java:511)
	at net.minecraft.client.Minecraft.func_147121_ag(Minecraft.java:1597)
	at net.minecraft.client.Minecraft.func_184117_aA(Minecraft.java:2254)
	at net.minecraft.client.Minecraft.func_184118_az(Minecraft.java:2020)

-- Affected level --
Details:
	Level name: MpServer
	All players: 1 total; [EntityPlayerSP['simon_kungen'/347, l='MpServer', x=15.74, y=40.00, z=16.44]]
	Chunk stats: MultiplayerChunkCache: 1089, 1089
	Level seed: 0
	Level generator: ID 01 - flat, ver 0. Features enabled: false
	Level generator options: 
	Level spawn location: World: (0,39,0), Chunk: (at 0,2,0 in 0,0; contains blocks 0,0,0 to 15,255,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Level time: 515752 game time, 13000 day time
	Level dimension: 0
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: false
	Forced entities: 2 total; [EntityDummy['entity.mountDummy.name'/346, l='MpServer', x=12.50, y=45.20, z=18.50], EntityPlayerSP['simon_kungen'/347, l='MpServer', x=15.74, y=40.00, z=16.44]]
	Retry entities: 0 total; []
	Server brand: fml,forge
	Server type: Integrated singleplayer server
Stacktrace:
	at net.minecraft.client.multiplayer.WorldClient.func_72914_a(WorldClient.java:420)
	at net.minecraft.client.Minecraft.func_71396_d(Minecraft.java:2741)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:427)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:196)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:231)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)

-- System Details --
Details:
	Minecraft Version: 1.12.2
	Operating System: Linux (amd64) version 5.4.0-74-generic
	Java Version: 1.8.0_291, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 5868970544 bytes (5597 MB) / 8256487424 bytes (7874 MB) up to 8256487424 bytes (7874 MB)
	JVM Flags: 2 total; -Xms4096m -Xmx8064m
	IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
	FML: MCP 9.42 Powered by Forge 14.23.5.2855 97 mods loaded, 97 mods active
	States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored

	| State  | ID                      | Version                         | Source                                              | Signature                                |
	|:------ |:----------------------- |:------------------------------- |:--------------------------------------------------- |:---------------------------------------- |
	| LCHIJA | minecraft               | 1.12.2                          | minecraft.jar                                       | None                                     |
	| LCHIJA | mcp                     | 9.42                            | minecraft.jar                                       | None                                     |
	| LCHIJA | FML                     | 8.0.99.99                       | forge-1.12.2-14.23.5.2855-universal.jar             | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| LCHIJA | forge                   | 14.23.5.2855                    | forge-1.12.2-14.23.5.2855-universal.jar             | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| LCHIJA | advancedrocketrycore    | 1                               | minecraft.jar                                       | None                                     |
	| LCHIJA | creativecoredummy       | 1.0.0                           | minecraft.jar                                       | None                                     |
	| LCHIJA | entity_culling          | 2.0.5                           | minecraft.jar                                       | None                                     |
	| LCHIJA | itemphysic              | 1.4.0                           | minecraft.jar                                       | None                                     |
	| LCHIJA | opencomputers|core      | 1.7.5.192                       | minecraft.jar                                       | None                                     |
	| LCHIJA | dsurroundcore           | 3.5.4.3                         | minecraft.jar                                       | None                                     |
	| LCHIJA | libvulpes               | 0.4.2.-83                       | LibVulpes-1.12.2-0.4.2-83-universal.jar             | None                                     |
	| LCHIJA | advancedrocketry        | @MAJOR@.@MINOR@.@REVIS@.@BUILD@ | AdvancedRocketry-1.12.2-2.0.0-248-universal.jar     | None                                     |
	| LCHIJA | akashictome             | 1.2-12                          | AkashicTome-1.2-12.jar                              | None                                     |
	| LCHIJA | appliedenergistics2     | rv6-stable-7                    | appliedenergistics2-rv6-stable-7.jar                | dfa4d3ac143316c6f32aa1a1beda1e34d42132e5 |
	| LCHIJA | carrots                 | 1.0.0b1                         | carrotslib-mc1.12.2-1.0.0b1.jar                     | None                                     |
	| LCHIJA | armorunder              | 1.0.0                           | armorunder-mc1.12.2-1.0.0.jar                       | None                                     |
	| LCHIJA | crafttweaker            | 4.1.20                          | CraftTweaker2-1.12-4.1.20.646.jar                   | None                                     |
	| LCHIJA | mtlib                   | 3.0.7                           | MTLib-3.0.7.jar                                     | None                                     |
	| LCHIJA | modtweaker              | 4.0.19                          | modtweaker-4.0.19.jar                               | None                                     |
	| LCHIJA | jei                     | 4.16.1.301                      | jei_1.12.2-4.16.1.301.jar                           | None                                     |
	| LCHIJA | quark                   | r1.6-179                        | Quark-r1.6-179.jar                                  | None                                     |
	| LCHIJA | autoreglib              | 1.3-32                          | AutoRegLib-1.3-32.jar                               | None                                     |
	| LCHIJA | opencomputers           | 1.7.5.192                       | OpenComputers-MC1.12.2-1.7.5.192.jar                | None                                     |
	| LCHIJA | batterybufferdriver     | 1.2                             | BatteryBufferDriver-1.2.jar                         | None                                     |
	| LCHIJA | baubles                 | 1.5.2                           | Baubles-1.12-1.5.2.jar                              | None                                     |
	| LCHIJA | betternether            | 0.1.8.6                         | betternether-0.1.8.6.jar                            | None                                     |
	| LCHIJA | bloodmoon               | 1.5.3                           | Bloodmoon-MC1.12.2-1.5.3.jar                        | d72e0dd57935b3e9476212aea0c0df352dd76291 |
	| LCHIJA | blur                    | 1.0.4-14                        | Blur-1.0.4-14.jar                                   | None                                     |
	| LCHIJA | bookshelf               | 2.3.577                         | Bookshelf-1.12.2-2.3.577.jar                        | d476d1b22b218a10d845928d1665d45fce301b27 |
	| LCHIJA | chameleon               | 1.12-4.1.3                      | Chameleon-1.12-4.1.3.jar                            | None                                     |
	| LCHIJA | chunkanimator           | 1.2                             | ChunkAnimator-MC1.12-1.2.jar                        | None                                     |
	| LCHIJA | codechickenlib          | 3.2.3.357                       | CodeChickenLib-1.12.2-3.2.3.357-universal.jar       | f1850c39b2516232a2108a7bd84d1cb5df93b261 |
	| LCHIJA | asielib                 | 1.0.0                           | Computronics-1.12.2-1.6.6.jar                       | None                                     |
	| LCHIJA | toughasnails            | 3.1.0.139                       | ToughAsNails-1.12.2-3.1.0.139-universal.jar         | None                                     |
	| LCHIJA | forestry                | 5.8.2.387                       | forestry_1.12.2-5.8.2.387.jar                       | None                                     |
	| LCHIJA | gregtech                | 1.16.1.753                      | gregtech-1.12.2-1.16.1.753.jar                      | None                                     |
	| LCHIJA | waila                   | 1.8.26                          | Hwyla-1.8.26-B41_1.12.2.jar                         | None                                     |
	| LCHIJA | tis3d                   | 1.5.1.37                        | TIS-3D-MC1.12.2-1.5.1.37.jar                        | None                                     |
	| LCHIJA | computronics            | 1.6.6                           | Computronics-1.12.2-1.6.6.jar                       | None                                     |
	| LCHIJA | consecration            | 1.0.6                           | consecration-1.12.2-1.0.6.jar                       | 2484ef4d131fdc0dca0647aa21b7b944ddb935a1 |
	| LCHIJA | ctgui                   | 1.0.0                           | CraftTweaker2-1.12-4.1.20.646.jar                   | None                                     |
	| LCHIJA | crafttweakerjei         | 2.0.3                           | CraftTweaker2-1.12-4.1.20.646.jar                   | None                                     |
	| LCHIJA | creativecore            | 1.10.0                          | CreativeCore_v1.10.61_mc1.12.2.jar                  | None                                     |
	| LCHIJA | customloadingscreen     | 1.12.2-1.5.1                    | CustomLoadingScreen-1.12.2-1.5.1.jar                | None                                     |
	| LCHIJA | cyberware               | 0.2.11.27                       | cyberware-1.12.2-0.2.11.27.jar                      | None                                     |
	| LCHIJA | damagetilt              | 0.1.0                           | DamageTilt-1.12.2-0.1.0.jar                         | None                                     |
	| LCHIJA | eleccore                | 1.9.452                         | ElecCore-1.12.2-1.9.452.jar                         | None                                     |
	| LCHIJA | mcjtylib_ng             | 3.5.0                           | mcjtylib-1.12-3.5.0.jar                             | None                                     |
	| LCHIJA | deepresonance           | 1.8.0                           | deepresonance-1.12-1.8.0.jar                        | None                                     |
	| LCHIJA | diethopper              | 1.1                             | diethopper-1.1.jar                                  | None                                     |
	| LCHIJA | fluidlogged_api         | 1.4                             | fluidlogged_api_v1.4_mc1.12.2.jar                   | None                                     |
	| LCHIJA | forgemultipartcbe       | 2.6.2.83                        | ForgeMultipart-1.12.2-2.6.2.83-universal.jar        | f1850c39b2516232a2108a7bd84d1cb5df93b261 |
	| LCHIJA | microblockcbe           | 2.6.2.83                        | ForgeMultipart-1.12.2-2.6.2.83-universal.jar        | None                                     |
	| LCHIJA | minecraftmultipartcbe   | 2.6.2.83                        | ForgeMultipart-1.12.2-2.6.2.83-universal.jar        | None                                     |
	| LCHIJA | gbook                   | 2.9.5                           | Guidebook-1.12.2-2.9.5.jar                          | None                                     |
	| LCHIJA | ichunutil               | 7.2.2                           | iChunUtil-1.12.2-7.2.2.jar                          | 4db5c2bd1b556f252a5b8b54b256d381b2a0a6b8 |
	| LCHIJA | itemscroller            | 0.12.0                          | itemscroller-1.12.2-0.12.0.jar                      | 2b03e1423915a189b8094816baa18f239d576dff |
	| LCHIJA | jeibees                 | 0.9.0.5                         | jeibees-0.9.0.5-mc1.12.2.jar                        | None                                     |
	| LCHIJA | journeymap              | 1.12.2-5.7.0                    | journeymap-1.12.2-5.7.0.jar                         | None                                     |
	| LCHIJA | mirage                  | 2.0                             | Mirage-2.0.3-rc3.jar                                | None                                     |
	| LCHIJA | mobdismemberment        | 7.0.0                           | MobDismemberment-1.12.2-7.0.0.jar                   | None                                     |
	| LCHIJA | mrtjpcore               | 2.1.4.43                        | MrTJPCore-1.12.2-2.1.4.43-universal.jar             | None                                     |
	| LCHIJA | norecipebook            | 1.2.1                           | noRecipeBook_v1.2.2formc1.12.2.jar                  | None                                     |
	| LCHIJA | nuclearcraft            | 2.18y                           | NuclearCraft-2.18y-1.12.2.jar                       | None                                     |
	| LCHIJA | overpoweredarmorbar     | @VERSION@                       | overloadedarmorbar-1.0.4g.jar                       | None                                     |
	| LCHIJA | particleculling         | 1.0                             | particleculling-1.12.2-v1.0.1.jar                   | None                                     |
	| LCHIJA | patchouli               | 1.0-20                          | Patchouli-1.0-20.jar                                | None                                     |
	| LCHIJA | peacefulsurface         | 1.12.2-v1b                      | PeacefulSurface-1.12.2-v1b.jar                      | aaaf83332a11df02406e9f266b1b65c1306f0f76 |
	| LCHIJA | performant              | 1.12.2-1.4                      | performant-1.12.2-1.10.jar                          | None                                     |
	| LCHIJA | thaumcraft              | 6.1.BETA26                      | Thaumcraft-1.12.2-6.1.BETA26.jar                    | None                                     |
	| LCHIJA | pneumaticcraft          | 1.12.2-0.11.15-398              | pneumaticcraft-repressurized-1.12.2-0.11.15-398.jar | None                                     |
	| LCHIJA | projectred-core         | 4.9.4.120                       | ProjectRed-1.12.2-4.9.4.120-Base.jar                | None                                     |
	| LCHIJA | projectred-integration  | 4.9.4.120                       | ProjectRed-1.12.2-4.9.4.120-integration.jar         | None                                     |
	| LCHIJA | projectred-transmission | 4.9.4.120                       | ProjectRed-1.12.2-4.9.4.120-integration.jar         | None                                     |
	| LCHIJA | redstoneflux            | 2.1.0                           | RedstoneFlux-1.12-2.1.0.6-universal.jar             | 8a6abf2cb9e141b866580d369ba6548732eff25f |
	| LCHIJA | rockhounding_core       | 3.27                            | rockhounding_core-1.12.2-3.27.jar                   | None                                     |
	| LCHIJA | rockhounding_chemistry  | 1.12.2-3.57                     | rockhounding_chemistry-1.12.2-3.57.jar              | None                                     |
	| LCHIJA | rsgauges                | 1.2.4                           | rsgauges-1.12.2-1.2.4.jar                           | ed58ed655893ced6280650866985abcae2bf7559 |
	| LCHIJA | savemystronghold        | 1.12.2-1.0.0                    | savemystronghold-1.12.2-1.0.0.jar                   | None                                     |
	| LCHIJA | sereneseasons           | 1.2.18                          | SereneSeasons-1.12.2-1.2.18-universal.jar           | None                                     |
	| LCHIJA | serenetweaks            | 0.0.1                           | SereneTweaks-1.0.jar                                | None                                     |
	| LCHIJA | sgcraft                 | 2.0.3                           | SGCraft-2.0.5.jar                                   | None                                     |
	| LCHIJA | soundfilters            | 0.12.1_for_1.12                 | SoundFilters-0.12.1_for_1.12.jar                    | None                                     |
	| LCHIJA | storagedrawers          | 1.12.2-5.4.0                    | StorageDrawers-1.12.2-5.4.0.jar                     | None                                     |
	| LCHIJA | tandimensions           | 0.3.0                           | tandimensions-0.3.0.jar                             | None                                     |
	| LCHIJA | thaumicaugmentation     | 1.12.2-2.1.1                    | ThaumicAugmentation-1.12.2-2.1.1.jar                | 8f678591ba6f78d579e553a8aa94b4c4766cb13d |
	| LCHIJA | thaumicjei              | 1.6.0                           | ThaumicJEI-1.12.2-1.6.0-27.jar                      | None                                     |
	| LCHIJA | thaumicenergistics      | 2.2.3                           | thaumicenergistics-2.2.4.jar                        | None                                     |
	| LCHIJA | travelersbackpack       | 1.0.35                          | TravelersBackpack-1.12.2-1.0.35.jar                 | None                                     |
	| LCHIJA | tbintegration           | 1.0.1                           | TravelersBackpackIntegration-1.12.2-1.0.1.jar       | None                                     |
	| LCHIJA | traverse                | 1.6.0                           | Traverse-1.12.2-1.6.0-69.jar                        | None                                     |
	| LCHIJA | wailaharvestability     | 1.1.12                          | WailaHarvestability-mc1.12-1.1.12.jar               | None                                     |
	| LCHIJA | worldstripper           | 1.6.0-1.12.2                    | World-Stripper-1.6.0-1.12.2.jar                     | None                                     |
	| LCHIJA | phosphor-lighting       | 1.12.2-0.2.6                    | phosphor-1.12.2-0.2.6+build50-universal.jar         | f0387d288626cc2d937daa504e74af570c52a2f1 |
	| LCHIJA | orelib                  | 3.5.2.2                         | OreLib-1.12.2-3.5.2.2.jar                           | 7a2128d395ad96ceb9d9030fbd41d035b435753a |
	| LCHIJA | dsurround               | 3.5.4.3                         | DynamicSurroundings-1.12.2-3.5.4.3.jar              | 7a2128d395ad96ceb9d9030fbd41d035b435753a |
	| LCHIJA | eleccoreloader          | 1.9.452                         | ElecCore-1.12.2-1.9.452.jar                         | None                                     |

	Loaded coremods (and transformers): 
CreativePatchingLoader (CreativeCore_v1.10.61_mc1.12.2.jar)
  
SSLoadingPlugin (SereneSeasons-1.12.2-1.2.18-universal.jar)
  sereneseasons.asm.transformer.EntityRendererTransformer
  sereneseasons.asm.transformer.WorldTransformer
TransformerLoader (OpenComputers-MC1.12.2-1.7.5.192.jar)
  li.cil.oc.common.asm.ClassTransformer
Quark Plugin (Quark-r1.6-179.jar)
  vazkii.quark.base.asm.ClassTransformer
PerformantLoadingPlugin (performant-1.12.2-1.10.jar)
  
LoadingPlugin (Bloodmoon-MC1.12.2-1.5.3.jar)
  lumien.bloodmoon.asm.ClassTransformer
GTCELoadingPlugin (gregtech-1.12.2-1.16.1.753.jar)
  gregtech.common.asm.GTCETransformer
PhosphorFMLLoadingPlugin (phosphor-1.12.2-0.2.6+build50-universal.jar)
  
ParticleCullingLoadingPlugin (particleculling-1.12.2-v1.0.1.jar)
  
Fluidlogged Plugin (fluidlogged_api_v1.4_mc1.12.2.jar)
  git.jbredwards.fluidlogged_api.asm.ASMHandler$Transformer
BlurPlugin (Blur-1.0.4-14.jar)
  com.tterrag.blur.BlurTransformer
ItemPatchingLoader (ItemPhysic_Full_1.4.34_mc1.12.2.jar)
  com.creativemd.itemphysic.ItemTransformer
Thaumic Augmentation Core Plugin (ThaumicAugmentation-1.12.2-2.1.1.jar)
  thecodex6824.thaumicaugmentation.core.TATransformer
ClsPlugin (CustomLoadingScreen-1.12.2-1.5.1.jar)
  alexiil.mc.mod.load.coremod.ClsTransformer
FMLPlugin (Mirage-2.0.3-rc3.jar)
  com.elytradev.mirage.asm.RenderGlobalTransformer
  com.elytradev.mirage.asm.ChunkRenderContainerTransformer
EntityCullingPlugin (EntityCulling-1.12.2-2.0.5.jar)
  meldexun.entityculling.plugin.ClassTransformer
DynamicSurroundingsCore (DynamicSurroundings-core-1.12.2-3.5.4.3.jar)
  org.orecruncher.dsurround.asm.Transformer
AdvancedRocketryPlugin (AdvancedRocketry-1.12.2-2.0.0-248-universal.jar)
  zmaster587.advancedRocketry.asm.ClassTransformer
LoadingPlugin (ChunkAnimator-MC1.12-1.2.jar)
  lumien.chunkanimator.asm.ClassTransformer
	GL info: ' Vendor: 'NVIDIA Corporation' Version: '4.6.0 NVIDIA 460.80' Renderer: 'GeForce GTX 660/PCIe/SSE2'
	AE2 Version: stable rv6-stable-7 for Forge 14.23.5.2768
	forestry : Modules have been disabled in the config: Mail, Better With Mods, EnderIO, Factory, Crate, Roots, Worktable, Charcoal, TechReborn, Energy, ImmersiveEngineering, Actually Additions, Mystical Agriculture
	AE2 Integration: IC2:OFF, RC:OFF, MFR:OFF, Waila:ON, InvTweaks:OFF, JEI:OFF, Mekanism:OFF, OpenComputers:ON, THE_ONE_PROBE:OFF, TESLA:OFF, CRAFTTWEAKER:ON
	Launched Version: MultiMC5
	LWJGL: 2.9.4
	OpenGL: GeForce GTX 660/PCIe/SSE2 GL version 4.6.0 NVIDIA 460.80, NVIDIA Corporation
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fml,forge'
	Type: Client (map_client.txt)
	Resource Packs: 
	Current Language: English (US)
	Profiler Position: N/A (disabled)
	CPU: 12x Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz
```
</details>

## 2
In vanilla Minecraft 1.13+ when a waterloggable block has two neighboring water sources said block will become waterlogged if possible. This feature is missing in your mod.

![2021-07-02_11 00 13](https://user-images.githubusercontent.com/9884180/124255517-874cd180-db2a-11eb-8a06-ad8b42f96ae3.png)

## 3
Right now when you fluidlogg a block it transforms it into a `TileEntity`, which is fine I guess as otherwise you'd have to add a new property for every new fluid in the game (~10 minutes load time + risk of data corruption). But this makes the block un-movable by Pistons. In vanilla, if you push a waterlogged block it destroys the water source in it. Perhaps you can have so if a (Sticky) Piston tries to push/drag a fluidlogged block it destroys the `TileEntity` and reverts it back to the regular block again which then gets moved.

**Comment by juraj-hrivnak** [comment]
For the **3rd** one, probably the simplest way to fix that is to use Quark.

**Comment by jbredwards** [comment]
1) This'll be fixed next update.
2) Good catch, will be added next update.
3) I probably won't implement your piston fluidlogged mechanic idea, since I think dispensers are a better solution. Through testing I found that dispensers can't place the fluid into blocks, this will be fixed next update. Also I can't add a new "fluidlogging property" for every fluid block in the game, that would be impossible. In 1.12, block properties are stored via an integer, this integer can only goes from 0-16. As you can imagine, this would cause some issues if I stored the block's id & the block's state this way.

**Comment by Shazuli** [comment]
> 
> 
> For the **3rd** one, probably the simplest way to fix that is to use Quark.

That was the first thing I tried:
![not_pushable](https://user-images.githubusercontent.com/9884180/124353101-b6d00c80-dc04-11eb-9b7f-4aa7e20861b9.gif)


And it seems like v1.4 has some issues compared to v1.3 @jbredwards that did not behave like this:
![2021-07-03_13 47 20](https://user-images.githubusercontent.com/9884180/124353198-65744d00-dc05-11eb-81dd-b8ba15cae188.png)

And another crash report (happens sometimes when you try to remove the fluid from the block with a bucket):
[crash-2021-07-03_13.47.49-client.txt](https://github.com/jbredwards/Fluidlogged-API/files/6758325/crash-2021-07-03_13.47.49-client.txt)

**Comment by jbredwards** [comment]
fixed all of the issues listed here in v1.5 :)
